src/utp_stream.cpp

```diff
@@ -1513,11 +1513,12 @@ bool utp_socket_impl::send_pkt(int const flags)
 
 		// did we fill up the whole mtu?
 		// if we didn't, we may still send it if there's
-		// no bytes in flight
+		// no undersized packet currently in flight
 		if (m_bytes_in_flight > 0
 			&& int(p->size) < std::min(int(p->allocated), effective_mtu)
 			&& !force
-			&& m_nagle)
+			&& m_nagle
+			&& !compare_less_wrap(m_nagle_seq_nr, m_acked_seq_nr, ACK_MASK))
```

Comment:
i `m_nagle_seq_nr` == `m_acked_seq_nr`, that means we don't have an outstanding undersized packet, right?
I believe this logic handles that case incorrectly. `compare_less_wrap()` would return false and we would hold off the packet.

If I understand correctly, this last condition should be:
```
&& compare_less_wrap(m_acked_seq_nr, m_nagle_seq_nr, ACK_MASK))
```